geomstats/geometry/positive_reals.py

```diff
@@ -0,0 +1,315 @@
+"""The manifold of positive reals.
+
+This manifold is a particular case in dimension 1
+of the SPD matrices endowed with the SPD affine-invariant metric
+with a power affine coefficient equal to one.
+This manifold is used as part of a product manifold involving
+complex Poincare disks to classify complex stationary centered
+Gaussian autoregressive time series in [Cabanes2022], [Cabanes_CESAR_2019],
+[Cabanes_GSI_2019] and [Cabanes_RADAR_2019].
+
+
+Lead author: Yann Cabanes.
+
+References
+----------
+.. [Cabanes2022] Yann Cabanes. Multidimensional complex stationary
+    centered Gaussian autoregressive time series machine learning
+    in Poincaré and Siegel disks: application for audio and radar
+    clutter classification, PhD thesis, 2022
+.. [Cabanes_CESAR_2019] Yann Cabanes, Frédéric Barbaresco, Marc Arnaudon,
+    Jérémie Bigot. Unsupervised Machine Learning for Pathological Radar
+    Clutter Clustering: the P-Mean-Shift Algorithm, C&ESAR 2019, Nov 2019,
+    Rennes, France. hal-02875430
+    https://hal.archives-ouvertes.fr/hal-02875430/document
+.. [Cabanes_GSI_2019] Yann Cabanes, Frédéric Barbaresco, Marc Arnaudon,
+    Jérémie Bigot. Toeplitz Hermitian Positive Definite Matrix Machine Learning
+    based on Fisher Metric. Geometric Science of Information, 2019,
+    10.1007/978-3-030-26980-7_27. hal-02875403
+    https://hal.archives-ouvertes.fr/hal-02875403/document
+.. [Cabanes_RADAR_2019] Yann Cabanes, Frédéric Barbaresco, Marc Arnaudon,
+    Jérémie Bigot. Non-Supervised High Resolution Doppler Machine Learning
+    for Pathological Radar Clutter. RADAR 2019, Sep 2020, Toulon, France.
+    10.1109/RADAR41533.2019.171295. hal-02875415
+    https://hal.archives-ouvertes.fr/hal-02875415/document
+.. [JV2016] B. Jeuris and R. Vandebril. The Kahler mean of Block-Toeplitz
+    matrices with Toeplitz structured blocks, 2016.
+    https://epubs.siam.org/doi/pdf/10.1137/15M102112X
+"""
+
+import geomstats.backend as gs
+from geomstats.geometry.base import OpenSet
+from geomstats.geometry.euclidean import Euclidean
+from geomstats.geometry.riemannian_metric import RiemannianMetric
+
+
+class PositiveReals(OpenSet):
+    """Class for the manifold of positive reals.
+
+    The real positive axis endowed with the Information geometry metric.
+
+    Parameters
+    ----------
+    scale : float
+        Scale of the positive reals metric.
+        Optional, default: 1.
+    """
+
+    def __init__(self, scale=1.0, **kwargs):
+        super().__init__(
+            dim=1,
+            embedding_space=Euclidean(1),
+            metric=PositiveRealsMetric(scale=scale),
+            **kwargs
+        )
+        self.scale = scale
+
+    @staticmethod
+    def belongs(point, atol=gs.atol):
```

Comment:
should we also allow `float`? I think so. If you agree, we should update the code in other places.

Reply:
Yes, we should allow both `float` and `complex`.
The natural data type of this manifold is `float`.
However, I have built this manifold the be part of a product manifold named `ProductPositiveRealsAndComplexPoincareDisks` whose natural data type is complex.
It is therefore very convenient if `PositiveReals` can work with both `float` and `complex` data types.

Reply:
What I mean by allowing `float` is allow this kind of code to work:

```python
space = PositiveReals()

space.belongs(2.)  #True
```

I don't agree we should take into consideration the complex case in this manifold (with exception of this function maybe). My reasoning is that we haven't considered complex cases in other (real) manifolds.

I know you want to use this in a `ProductManifold` in combination with complex manifolds. I think it should be the `ProductManifold` taking care of this, not the (real) manifold.